scripts/build_desktop_app_with_firebase.py

```diff
@@ -0,0 +1,281 @@
+Build successful!
+Please find your executables in build directory: /Users/<user>/quickstart-cpp/database/testapp/build_source
+
+# Running the built example
+$ ./Users/<user>/quickstart-cpp/database/testapp/build_source/desktop_testapp
+"""
+import argparse
+import distutils.spawn
+import os
+import platform
+import subprocess
+import sys
+
+def is_path_valid_for_cmake(path):
+  """Check if specified path is setup for cmake"""
+  return os.path.exists(os.path.join(path, 'CMakeLists.txt'))
+
+def is_sdk_path_source(sdk_dir):
+  """Validate if firebase sdk dir is firebase cpp source dir"""
+  # Not the most reliable way to detect if the sdk path is source or prebuilt but
+  # should work for our purpose.
+  return os.path.exists(os.path.join(sdk_dir, 'build_tools'))
+
+def validate_prebuilt_args(arch, config):
+  """Validate cmd line args for build with prebuilt libraries"""
+  # Some options are not available when using prebuilt libraries"""
+  if platform.system() == 'Darwin':
+    if arch == 'x86' or config == 'Debug':
+      raise ValueError("Prebuilt mac firebase libraries are built for x64 and Release mode only. "
+                       "Please fix the command line arguments and try again")
+
+  if platform.system() == 'Linux':
+    if config == 'Debug':
+      raise ValueError("Prebuilt linux firebase libraries are built with Release mode only. "
+                       "Please fix the --config command line argument and try again.")
+
+def get_vcpkg_triplet(arch, msvc_runtime_library='static'):
+  """ Get vcpkg target triplet (platform definition).
+
+  Args:
+    arch (str): Architecture (eg: 'x86', 'x64').
+    msvc_runtime_library (str): Runtime library for MSVC (eg: 'static', 'dynamic').
+
+  Raises:
+    ValueError: If current OS is not win,mac or linux.
+
+  Returns:
+    (str): Triplet name.
+       Eg: "x64-windows-static".
+  """
+  triplet_name = [arch]
+  if platform.system() == 'Windows':
+    triplet_name.append('windows')
+    triplet_name.append('static')
+    if msvc_runtime_library == 'dynamic':
+      triplet_name.append('md')
+  elif platform.system() == 'Darwin':
+    triplet_name.append('osx')
+  elif platform.system() == 'Linux':
+    triplet_name.append('linux')
+  else:
+    raise ValueError("Unsupported platform. "
+                     "This function works only on Windows, Linux or Mac platforms.")
+
+  triplet_name = '-'.join(triplet_name)
+  print("Using vcpkg triplet: {0}".format(triplet_name))
+  return triplet_name
+
+def build_source_vcpkg_dependencies(sdk_source_dir, arch, msvc_runtime_library):
+  """Build C++ dependencies for firebase source SDK using vcpkg.
+
+  Args:
+   sdk_source_dir (str): Path to Firebase C++ source directory.
+   arch (str): Platform Architecture (example: 'x64').
+   msvc_runtime_library (str): Runtime library for MSVC (eg: 'static', 'dynamic').
+  """
+  # TODO: Remove this once dev branch of firebase-cpp-sdk repo has been merged
```

Comment:
Recommend creating a ticket on the backlog  and referencing it here.

Reply:
Great idea. Done!